cmd/config/identity/ldap/config.go

```diff
@@ -422,7 +432,13 @@ func Lookup(kvs config.KVS, rootCAs *x509.CertPool) (l Config, err error) {
 	// Lookup bind user configuration
 	lookupBindDN := env.Get(EnvLookupBindDN, kvs.Get(LookupBindDN))
 	lookupBindPassword := env.Get(EnvLookupBindPassword, kvs.Get(LookupBindPassword))
-	if lookupBindDN != "" && lookupBindPassword != "" {
+	if v := env.Get(EnvLookupUnauthenticated, kvs.Get(LookupUnauthenticated)); v != "" {
+		l.LookupUnauthenticated, err = config.ParseBool(v)
+		if err != nil {
+			return l, err
+		}
+	}
+	if (lookupBindDN != "" && lookupBindPassword != "") || l.LookupUnauthenticated {
```

Comment:
This would mean that lookupBindDN is optional, that is not the case here and let's not make it optional here even if the SPEC allows it because unauthenticated is already a bad practice for LDAP lookups as mentioned due to underlying security issues (https://tools.ietf.org/html/rfc4513#section-5.1.2) and we should allow these requests to be traced/audited on the server - so enforcing client to setup a BIND_DN is essential for audit purposes.
 
This condition should be changed as 

```suggestion
	if lookupBindDN != "" && (lookupBindPassword != "" || l.LookupUnauthenticated) {
```